### DISCUSSION TITLE
Seeking help in reconnection attempt - Android SDK

### DISCUSSION THREAD

**Discussion by AndroidApprikart** [ant-media/Ant-Media-Server#5588] [Q&A]
With the updated Android SDK, we are exploring the track-based conferencing solution. For this, we are exploring the sample app's MultiTrackConferenceActivtiy.java

Feature to be implemented: For a participant in the VC Room, if the user gets disconnected without the consent(due to network or any internal error)..a rejoin option must be given. And when the user rejoins the the session and chat messages must be restored.

In the current SDK, can see a onReconnectionAttempt() call back in the IWebRTCListener interface. This gets called even when there is strong  network connection. 

Can anyone help me in understanding the reconnect scenario implemented in the current SDK?

**Comment by burak-58** [comment]
Hi @AndroidApprikart ,
Thanks for the report.
Firstly I strongly recommend using TrackBasedConferenceActivity.java instead of deprecated MultiTrackConferenceActivtiy.java.

For the problem (getting onReconnectionAttempt after reconnection), I will check it and if I can reproduce then I will convert this discussion to an issue.

**Reply by AndroidApprikart** [reply]
For using TrackBasedConferenceActivity: As per [https://github.com/orgs/ant-media/discussions/5534](this), I had reported crash in TrackBasedConferenceActivity.java. 

The scenario is as follows:

 When 2 participants have successfully joined the room, if the publishing participant turns off the camera the application crashes. Below is the error log observed.

FATAL EXCEPTION: pool-3-thread-1
                                         Process: io.antmedia.webrtc_android_sample_app, PID: 11042
                                         java.lang.IllegalStateException: MediaStreamTrack has been disposed.
                                         	at org.webrtc.MediaStreamTrack.checkMediaStreamTrackExists(MediaStreamTrack.java:120)
                                         	at org.webrtc.MediaStreamTrack.setEnabled(MediaStreamTrack.java:98)
                                         	at io.antmedia.webrtcandroidframework.WebRTCClient.lambda$setVideoEnabled$35$io-antmedia-webrtcandroidframework-WebRTCClient(WebRTCClient.java:2488)
                                         	at io.antmedia.webrtcandroidframework.WebRTCClient$$ExternalSyntheticLambda34.run(Unknown Source:4)
                                         	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
                                         	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
                                         	at java.lang.Thread.run(Thread.java:1012)

For this I was advised to use the MultitrackConferenceActivity. What is the way forward now?

For reconnection, could you please help me out if not through SDK, how can I handle in the app level.

**Reply by KavyashreeBasavesh** [reply]
@AndroidApprikart 
Please let me know is it working for you or not. Even, I am facing the same issue.

@burak-58 @yashtandon113 Requesting your help in resolving this issue.

**Reply by KavyashreeBasavesh** [reply]
@burak-58 
Hi Burak,
Can you please confirm, if the multitrackconferencemanager is available in the latest SDK which was updated yesterday.
If yes, how can we proceed further with the issue we are facing with multitrackconference.

**Reply by burak-58** [reply]
Hi @KavyashreeBasavesh,
In the new SDK, there is one Conference sample Activity. Please check it. There is no more `MultiTrackConferenceManger`. `WebRTCClient `handles peer connections for both publishing and playing. Also one important class is [DefaultConferenceWebRTCListener](https://github.com/ant-media/WebRTC-Android-SDK/blob/master/webrtc-android-framework/src/main/java/io/antmedia/webrtcandroidframework/api/DefaultConferenceWebRTCListener.java) which is a bridge between `WebRTCClient` and Activity. Also you can extend it as you want.
